tests/integration/util.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.
+
+package integration
+
+import (
+	"github.com/pulumi/pulumi/pkg/v3/engine"
+	"github.com/pulumi/pulumi/pkg/v3/testing/integration"
+	"os"
+	"testing"
+)
+
+func getCwd(t *testing.T) string {
+	cwd, err := os.Getwd()
+	if err != nil {
+		t.FailNow()
+	}
+	return cwd
+}
+
+func getJavaBase(t *testing.T, testSpecificOptions integration.ProgramTestOptions) integration.ProgramTestOptions {
+	opts := integration.ProgramTestOptions{
+		Config: map[string]string{
+			"org": "pulumi-bot",
+		},
+		PrepareProject: func(*engine.Projinfo) error {
+			return nil // needed because defaultPrepareProject does not know about java
+		},
+	}
+	opts = opts.With(testSpecificOptions)
+
+	// local environment, to run locally offline, make sure you set:
```

Comment:
This is nice! TBH I would support just switching to file-state backend completely in CI also. Java is a new addition to the Pulumi ecosystem and we do not need to test the cloud backend per se, and we are pretty sure the kinds of bugs we're trying to find will be found using filestate backend anyway. I don't suggest we make edits here I'll just make a ticket. https://github.com/pulumi/pulumi-java/issues/811

Reply:
This code was not added in this PR, I just moved these functions. I definitely agree with your point, though.